### PR TITLE
Refactor PoolingLayerTest

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Refactor PoolingLayerTest

### Tickets:
 - [CVS-111364](https://jira.devtools.intel.com/browse/CVS-111364)

### CODE REVIEW

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/pooling.hpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#include <string>
+#include <vector>
+
+#include "shared_test_classes/base/ov_subgraph.hpp"
+#include "common_test_utils/test_enums.hpp"
+
+namespace ov {
+namespace test {
+typedef std::tuple<
+        ov::test::utils::PoolingTypes,  // Pooling type, max or avg
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/pooling.hpp:17]: Do we have the same enum in ov? if yes, please replace by OV

File: src/tests/functional/shared_test_classes/src/single_op/pooling.cpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    std::vector<size_t> pad_begin, pad_end;
+    ov::op::PadType pad_type;
+    ov::op::RoundingType rounding_type;
+    bool excludePad;
+    std::tie(pool_type, kernel, stride, pad_begin, pad_end, rounding_type, pad_type, excludePad) = pool_params;
+    init_input_shapes(shapes);
+
+    auto param = std::make_shared<ov::op::v0::Parameter>(model_type, inputDynamicShapes.front());
+
+    std::shared_ptr<ov::Node> pooling = ngraph::builder::makePooling(param,
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/single_op/pooling.cpp:81]: `ngraph`?

Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/single_op/pooling.cpp:81]: Builders are still in ngraph namespace. Move them after test refactoring

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/pooling.hpp (hunk 1 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/pooling.hpp:17]: What you mean? It is already in ov namespace (common_test_utils/test_enums.hpp)

Comment (iefode) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/pooling.hpp:17]: I mean not test namespace. I think we should have same types in OV lib

Comment (olpipi) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/pooling.hpp:17]: I checked, there are no PoolingTypes in main lib now.

**Comment by vurusovs** [comment]
Closing due changes are merged in scope of https://github.com/openvinotoolkit/openvino/pull/20495

**Pull request closed without merging**